sim/druid/druid.go

```diff
@@ -227,6 +229,39 @@ func (druid *Druid) RegisterFeralTankSpells() {
 	// druid.registerSwipeBearSpell()
 }
 
+// Override the spell ApplyEffects functions for any consumables that require the druid to unshift out of Cat or Bear form
+// in order to be used so that the druid unshifts before usage.
+func (druid *Druid) handleConsumablesUnshifts() {
+	druid.RegisterAura(core.Aura{
+		Label:      "Apply clear form to consumes",
+		Duration:   core.NeverExpires,
+		BuildPhase: core.CharacterBuildPhaseConsumes,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			for _, cd := range druid.GetMajorCooldowns() {
+				if cd != nil && cd.Spell.Flags.Matches(core.SpellFlagNoShapeshift) {
+					initialMCD := druid.GetInitialMajorCooldown(cd.Spell.ActionID)
+
+					oldApplyEffects := cd.Spell.ApplyEffects
+					newApplyEffects := func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+						druid.ClearForm(sim)
+						oldApplyEffects(sim, target, spell)
+					}
+					cd.Spell.ApplyEffects = newApplyEffects
+					initialMCD.Spell.ApplyEffects = newApplyEffects
+
+					oldShouldActivate := cd.ShouldActivate
+					newShouldActivate := func(sim *core.Simulation, character *core.Character) bool {
+						return oldShouldActivate(sim, character) && !druid.form.Matches(Bear|Cat)
+					}
+					cd.ShouldActivate = newShouldActivate
+					initialMCD.ShouldActivate = newShouldActivate
+				}
+			}
+		},
+	})
```

Comment:
Despite these overrides and confirming that potions and explosives are being found, the applied function overrides are seemingly not working. I added print statements within the new function definitions to confirm this